torchgeo/datasets/millionaid.py

```diff
@@ -0,0 +1,397 @@
+        "bridge",
+        "cemetery",
+        "church",
+        "commercial_area",
+        "commercial_land",
+        "dam",
+        "desert",
+        "detached_house",
+        "dry_field",
+        "factory_area",
+        "forest",
+        "golf_course",
+        "grassland",
+        "greenhouse",
+        "ground_track_field",
+        "helipad",
+        "highway_area",
+        "ice_land",
+        "industrial_land",
+        "intersection",
+        "island",
+        "lake",
+        "leisure_land",
+        "meadow",
+        "mine",
+        "mining_area",
+        "mobile_home_park",
+        "oil_field",
+        "orchard",
+        "paddy_field",
+        "parking_lot",
+        "pier",
+        "port_area",
+        "power_station",
+        "public_service_land",
+        "quarry",
+        "railway",
+        "railway_area",
+        "religious_land",
+        "residential_land",
+        "river",
+        "road",
+        "rock_land",
+        "roundabout",
+        "runway",
+        "solar_power_plant",
+        "sparse_shrub_land",
+        "special_land",
+        "sports_land",
+        "stadium",
+        "storage_tank",
+        "substation",
+        "swimming_pool",
+        "tennis_court",
+        "terraced_field",
+        "train_station",
+        "transportation_land",
+        "unutilized_land",
+        "viaduct",
+        "wastewater_plant",
+        "water_area",
+        "wind_turbine",
+        "woodland",
+        "works",
+    ]
+
+    multi_class_categories = [
+        "apartment",
+        "apron",
+        "bare_land",
+        "baseball_field",
+        "bapsketball_court",
+        "beach",
+        "bridge",
+        "cemetery",
+        "church",
+        "commercial_area",
+        "dam",
+        "desert",
+        "detached_house",
+        "dry_field",
+        "forest",
+        "golf_course",
+        "greenhouse",
+        "ground_track_field",
+        "helipad",
+        "ice_land",
+        "intersection",
+        "island",
+        "lake",
+        "meadow",
+        "mine",
+        "mobile_home_park",
+        "oil_field",
+        "orchard",
+        "paddy_field",
+        "parking_lot",
+        "pier",
+        "quarry",
+        "railway",
+        "river",
+        "road",
+        "rock_land",
+        "roundabout",
+        "runway",
+        "solar_power_plant",
+        "sparse_shrub_land",
+        "stadium",
+        "storage_tank",
+        "substation",
+        "swimming_pool",
+        "tennis_court",
+        "terraced_field",
+        "train_station",
+        "viaduct",
+        "wastewater_plant",
+        "wind_turbine",
+        "works",
+    ]
+
+    md5s = {
+        "train": "1b40503cafa9b0601653ca36cd788852",
+        "test": "https://eastus1-mediap.svc.ms/transform/zip?cs=fFNQTw",
+    }
+    url = {
+        "train": "https://eastus1-mediap.svc.ms/transform/zip?cs=fFNQTw",
+        "test": "https://eastus1-mediap.svc.ms/transform/zip?cs=fFNQTw",
```

Comment:
Does this actually download the test set? These look like the same url to me. Also looks like the test set is made of multiple parts (e.g. `test.zip.001`, `test.zip.002`, etc.). Does `extract_archive` support this?

Reply:
That is a good question, I'm not sure whether that will work or not. We've had to hack things to support deflate64-compressed zip files before, so multi-part zip files should be possible somehow.

Reply:
> Does this actually download the test set? These look like the same url to me.

If I look in the [OneDrive](https://whueducn-my.sharepoint.com/personal/longyang_whu_edu_cn/_layouts/15/onedrive.aspx?ga=1&id=%2Fpersonal%2Flongyang%5Fwhu%5Fedu%5Fcn%2FDocuments%2FMillion%2DAID%2FMillion%2DAID) and download the train or test folder, the download link for me are the same for some reason.

Reply:
@isaaccorley you mentioned that you downloaded the test set and computed its MD5. Where did you download the test set from? Was it multiple parts when you downloaded and checksummed it?

Reply:
@isaaccorley you mentioned that you downloaded the test set and computed its MD5. Where did you download the test set from? Was it multiple parts when you downloaded and checksummed it?

Reply:
I used the OneDrive link above and right clicked on the `test` folder and it downloaded as `test.zip` which I then took the md5 of. The test set has multiple parts so even after you extract `test.zip` you still need to extract them together.

Reply:
I actually just noticed that `test.zip` was corrupted and I couldn't unzip it completely so I think the hash may be incorrect. Going to try and download each of the test files individually.

Reply:
Was this resolved?

Reply:
I don't think this was ever resolved, this is the only thing holding up this PR. If there's a single link for all data that's fine, we can download the data with one link, checksum it once, then extract it and extract and zip files that contains. I don't have the bandwidth/storage to download this myself, but can someone investigate this and see if it's even possible to download this dataset? If not, we could just remove the download logic until we figure it out.

Reply:
Did anyone ever reach out to the dataset authors and see if we can rehost a single zip file on something like zenodo?